### PR TITLE
Mc provider annotations

### PR DESCRIPTION
**Pull request by X52p**
Adds the possibility of using Java Annotations for:
Actions (@Action for Method) with parameters (@ActionParameter)
Aggregations (@Aggregation for class)
Parameters (@Parameter for field)

Implementations without Annotations are NOT effected by this.
Hybrid Implementations are possible.
(See PayloadsTestMCAddapter.java for example)

### CODE REVIEW

File: core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/ActionParameter.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
```
Comment (dmarszk) [inline core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/ActionParameter.java:1]: Add NMF License header

File: core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/Aggregation.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
```
Comment (dmarszk) [inline core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/Aggregation.java:2]: Add NMF License header

File: core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/Aggregations.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
```
Comment (dmarszk) [inline core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/Aggregations.java:2]: Add NMF License header

**Comment by dmarszk** [comment]
Could you add an example of using `@aggregation` annotation to the built-in apps?

File: core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/Aggregations.java (hunk 3 above)
Comment (X52p) [reply core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/Aggregations.java]: done

File: core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/Aggregation.java (hunk 2 above)
Comment (X52p) [reply core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/Aggregation.java:2]: done

File: core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/ActionParameter.java (hunk 1 above)
Comment (X52p) [reply core/nmf-composites/generic-composite-model/src/main/java/esa/mo/nmf/annotations/ActionParameter.java]: done

**Pull request merged**